integration-tests/src/test/java/oracle/weblogic/kubernetes/ItDiagnosticsCompleteAvailableCondition.java

```diff
@@ -0,0 +1,745 @@
+   */
+  @Disabled
+  @Order(4)
+  @Test
+  @DisplayName("Test domain status condition with cluster replica set to zero and min-replicas set to zero")
+  void testCompleteAvailableConditionWithReplicaZero() {
+    String patchStr;
+    try {
+      logger.info("patch the domain resource with new cluster replica 0");
+      patchStr = "[{\"op\": \"replace\",\"path\": \"/spec/clusters/0/replicas\", \"value\": 0}]";
+
+      logger.info("Updating domain configuration using patch string: {0}", patchStr);
+      assertTrue(patchDomainCustomResource(domainUid, domainNamespace1, new V1Patch(patchStr),
+          V1Patch.PATCH_FORMAT_JSON_PATCH), "Failed to patch domain");
+
+      // verify the admin server service exists
+      checkPodReadyAndServiceExists(adminServerPodName, domainUid, domainNamespace1);
+
+      // verify the cluster server pods are shutdown
+      logger.info("Checking managed server pods were shutdown");
+      for (int i = 1; i <= replicaCount; i++) {
+        checkPodDoesNotExist(managedServerPodNamePrefix + i, domainUid, domainNamespace1);
+      }
+
+      // verify the condition type Completed exists
+      checkDomainStatusConditionTypeExists(domainUid, domainNamespace1, DOMAIN_STATUS_CONDITION_COMPLETED_TYPE);
+      // verify the condition type Available exists
+      checkDomainStatusConditionTypeExists(domainUid, domainNamespace1, DOMAIN_STATUS_CONDITION_AVAILABLE_TYPE);
+      // verify the condition Completed type has status True
+      checkDomainStatusConditionTypeHasExpectedStatus(domainUid, domainNamespace1,
+          DOMAIN_STATUS_CONDITION_COMPLETED_TYPE, "True");
+      // verify the condition Available type has status True
+      checkDomainStatusConditionTypeHasExpectedStatus(domainUid, domainNamespace1,
+          DOMAIN_STATUS_CONDITION_AVAILABLE_TYPE, "False");
+      // verify there is no status condition type Failed
+      verifyDomainStatusConditionTypeDoesNotExist(domainUid, domainNamespace1, DOMAIN_STATUS_CONDITION_FAILED_TYPE);
+
+    } finally {
+      // patch the domain back to the original state
+      logger.info("patch the domain and change the cluster replicas to 2");
+      patchStr = "[{\"op\": \"replace\",\"path\": \"/spec/clusters/0/replicas\", \"value\": 2}]";
+
+      logger.info("Updating domain configuration using patch string: {0}", patchStr);
+      assertTrue(patchDomainCustomResource(domainUid, domainNamespace1, new V1Patch(patchStr),
+          V1Patch.PATCH_FORMAT_JSON_PATCH), "Failed to patch domain");
+
+      logger.info("Checking for admin server pod is up and running");
+      checkPodReadyAndServiceExists(adminServerPodName, domainUid, domainNamespace1);
+
+      // verify all the managed servers are up and running
+      logger.info("Checking managed server pods are up and running");
+      for (int i = 1; i <= replicaCount; i++) {
+        checkPodReadyAndServiceExists(managedServerPodNamePrefix + i, domainUid, domainNamespace1);
+      }
+    }
+  }
```

Comment:
Can you move the code in finally block into a method say resetDomainResource() to avoid the code repetition. 

By resetting the domain Status in finally block, do we still need method Order

Reply:
done